tests/test_player.py

```diff
@@ -0,0 +1,321 @@
+        assert len(result) == 3
+        dx, dy, dz = result
+        assert math.isclose(dx, 1e-15, abs_tol=1e-15)
+        assert dy == 0
+        assert math.isclose(dz, 1)
+
+    def test_get_movement_vector_right_movement(self, player):
+        player.flying = False
+        player.strafe = [0, 1]
+        result = player.get_motion_vector()
+        assert len(result) == 3
+        dx, dy, dz = result
+        assert math.isclose(dx, 1)
+        assert dy == 0
+        assert math.isclose(dz, 1e-15, abs_tol=1e-15)
+
+    def test_get_movement_vector_left_movement(self, player):
+        player.flying = False
+        player.strafe = [0, -1]
+        result = player.get_motion_vector()
+        assert len(result) == 3
+        dx, dy, dz = result
+        assert math.isclose(dx, -1)
+        assert dy == 0
+        assert math.isclose(dz, 1e-15, abs_tol=1e-15)
+
+    def test_get_movement_vector_forward_movement_with_flight(self, player):
+        player.flying = True
+        player.strafe = [-1, 0]
+        result = player.get_motion_vector()
+        assert len(result) == 3
+        dx, dy, dz = result
+        assert math.isclose(dx, 1e-15, abs_tol=1e-15)
+        assert dy == 0
+        assert math.isclose(dz, -1)
+
+    def test_get_movement_vector_backwward_movement_with_flight(self, player):
+        player.flying = True
+        player.strafe = [1, 0]
+        result = player.get_motion_vector()
+        assert len(result) == 3
+        dx, dy, dz = result
+        assert math.isclose(dx, 1e-15, abs_tol=1e-15)
+        assert dy == 0
+        assert math.isclose(dz, 1)
+
+    def test_get_movement_vector_right_movement_with_flight(self, player):
+        player.flying = True
+        player.strafe = [0, 1]
+        result = player.get_motion_vector()
+        assert len(result) == 3
+        dx, dy, dz = result
+        assert math.isclose(dx, 1)
+        assert dy == 0
+        assert math.isclose(dz, 1e-15, abs_tol=1e-15)
+
+    def test_get_movement_vector_left_movement_with_flight(self, player):
+        player.flying = True
+        player.strafe = [0, -1]
+        result = player.get_motion_vector()
+        assert len(result) == 3
+        dx, dy, dz = result
+        assert math.isclose(dx, -1)
+        assert dy == 0
+        assert math.isclose(dz, 1e-15, abs_tol=1e-15)
+
+    def test_speed_up_no_call_should_be_default_speed(self, player):
+        assert player.walking_speed == player.WALK_SPEED_INCREMENT
+
+    def test_speed_up_called_one_time_should_be_double_initial_speed(self, player):
+        player.speed_up()
+        assert player.walking_speed == 2 * player.WALK_SPEED_INCREMENT
+
+    def test_speed_up_called_two_times_should_be_triple_initial_speed(self, player):
+        player.speed_up()
+        player.speed_up()
+        assert player.walking_speed == 3 * player.WALK_SPEED_INCREMENT
+
+    def test_speed_up_called_three_times_should_be_quadruple_initial_speed(self, player):
```

Comment:
A loop will make it easier and fewer lines!
speed step by 5 depends on def speed_up & speed_down.

Reply:
Yes, a loop would make it fewer lines; however the intent with those tests is to be *explicit* with how many calls are being made to `speed_up` and `speed_down` to illustrate what happens, such as calling speed up 4 or more times, caps the speed out at 4x WALK_SPEED_INCREMENT, aka: 20.

Reply:
I take it